src/hooks/safe-apps/useRemoveAppModal.ts

```diff
@@ -0,0 +1,28 @@
+import { useState, useCallback } from 'react'
+import { SafeAppData } from '@gnosis.pm/safe-react-gateway-sdk'
+
+type ModalState =
+  | {
+      isOpen: false
+      app: null
+    }
+  | {
+      isOpen: true
+      app: SafeAppData
```

Comment:
Perhaps something cooler can be done here with [conditional types](https://www.typescriptlang.org/docs/handbook/2/conditional-types.html)?

Reply:
This type doesn't take an input type, so I don't know if it's applicable here. In general, yeah I wonder if there's a better way to do this

Reply:
the old typescript handbook had a somewhat similar example: https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions

But I can't find any reference to this pattern in the new docs

Reply:
I've just discovered that the pattern I've used here is called `Discriminating Union` 😎